libs/notion/src/lib/notion.ts

```
@@ -0,0 +1,112 @@
+      count++;
+    }
+    rows.push(newRow);
+  }
+  return {
+    fields: notionFields,
+    rows: rows,
+  };
+}
+
+export function typeTranslationNotionToGeoJson(notionData: object): string {
+  // Might be more complete later ?
+  if (notionData['type'] === 'number') {
+    return 'real';
+  }
+  return 'string';
+}
+
+export function extractContentOfNotionField(jsonContent: { type }): string | string[] {
+  if (jsonContent.type == null || jsonContent.type == 'null') {
+    return '';
+  }
+  if (jsonContent.type === 'title') {
+    return jsonContent[jsonContent.type][0].plain_text;
+  }
+  if (jsonContent.type === 'email') {
+    return jsonContent[jsonContent.type];
+  }
+  if (jsonContent.type === 'number') {
+    return jsonContent[jsonContent.type];
+  }
+  if (jsonContent.type === 'url') {
+    return jsonContent[jsonContent.type];
+  }
+  if (jsonContent.type === 'date') {
+    return jsonContent[jsonContent.type].start;
+  }
+  if (jsonContent.type === 'created_time') {
+    return jsonContent[jsonContent.type];
+  }
+  if (jsonContent.type === 'phone_number') {
+    return jsonContent[jsonContent.type];
+  }
+
+  if (jsonContent.type === 'rich_text') {
+    /*
+        RICH TEXTS COULD BE ANYTHING !
+        I CHOSE TO EXTRACT DATA THIS WAY :
+        --> [] -> returns ''
+        --> huge object with href not empty -> returns href content
+        --> in other cases -> returns plain_text content
+     */
+    if (jsonContent[jsonContent.type].length === 0) {
+      return '';
+    } else if (
+      jsonContent[jsonContent.type][0].href != '' &&
+      jsonContent[jsonContent.type][0].href != 'null' &&
+      jsonContent[jsonContent.type][0].href != null
+    ) {
+      return jsonContent[jsonContent.type][0].href;
+    }
+    return jsonContent[jsonContent.type][0].plain_text;
+  }
+  if (jsonContent.type === 'select') {
+    if (jsonContent[jsonContent.type] !== null) {
+      return jsonContent[jsonContent.type].name;
+    }
+    return '';
```

In most unexpected use cases you might `throw` an error instead of returning an empty string.

---

Or not if this is expected behavior !